internal/driver/driver.go

```diff
@@ -121,12 +133,51 @@ func (d *Driver) Initialize(sdk interfaces.DeviceServiceSDK) error {
 
 	// Make sure the paths of existing devices are up-to-date.
 	go d.RefreshExistingDevicePaths()
-
+	go d.StartRtspCredentialServer()
 	return nil
 }
+func (d *Driver) StartRtspCredentialServer() {
+	d.lc.Infof("starting rtsp server")
+	rtspAuthServer := http.NewServeMux()
+	rtspAuthServer.HandleFunc("/rtspauth", d.RTSPCredentialsHandler)
+	http.ListenAndServe(d.rtspAuthenticationServer, rtspAuthServer)
+}
+func (d *Driver) RTSPCredentialsHandler(w http.ResponseWriter, r *http.Request) {
 
-func (d *Driver) Start() error {
-	return nil
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		d.lc.Errorf("could not read body: %s", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	var rtspAuthRequest RTSPAuthRequest
+	err = json.Unmarshal(body, &rtspAuthRequest)
+	if err != nil {
+		d.lc.Errorf("could not unmarshal read body: %s", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if rtspAuthRequest.User == "" || rtspAuthRequest.Password == "" {
+		d.lc.Debug("username or password is empty") // this can happen during normal operation
+		w.WriteHeader(http.StatusBadRequest)
+	}
+	credential, edgexErr := d.tryGetCredentials(rtspAuthKey)
+	if edgexErr != nil {
+		d.lc.Warnf("Failed to retrieve credentials for rtsp authentication from the secret store. Have you stored credentials yet for secretName %s?", rtspAuthKey)
+		w.WriteHeader(http.StatusUnauthorized)
```

Comment:
I think we should write in the response body that there is a problem with the configuration. this could help user debug it if they forget to set it up.